kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/modem/quectel/generic/QuectelGeneric.java

```diff
@@ -165,6 +173,131 @@ public ModemRegistrationStatus getRegistrationStatus() throws KuraException {
     }
 
+    public void getQueryNetworkInformation() throws KuraException {
+        synchronized (this.atLock) {
+            String atPort = getAtPort();
+
+            logger.debug("sendCommand getQueryNetworkInformation :: {}", QuectelGenericAtCommands.GET_QUERY_NETWORK_INFORMATION.getCommand());
+            byte[] reply;
+            CommConnection commAtConnection = openSerialPort(atPort);
+            if (!isAtReachable(commAtConnection)) {
+                closeSerialPort(commAtConnection);
+                throw new KuraException(KuraErrorCode.NOT_CONNECTED, MODEM_NOT_AVAILABLE_FOR_AT_CMDS_MSG);
+            }
+            try {
+                reply = commAtConnection.sendCommand(QuectelGenericAtCommands.GET_QUERY_NETWORK_INFORMATION.getCommand().getBytes(),
+                        1000, 100);
+            } catch (IOException e) {
+                closeSerialPort(commAtConnection);
+                throw new KuraException(KuraErrorCode.CONNECTION_FAILED, e);
+            }
+            closeSerialPort(commAtConnection);
+            if (reply != null) {
+                String sQnwinfo = this.getResponseString(reply);
+                getQueryNetworkInformationReply(sQnwinfo);
+            }
+        }
+    }
+
+    public String[] getQueryNetworkInformationReply(String sQnwinfo) throws KuraException {
+        if (sQnwinfo.startsWith("+QNWINFO:")) {
+            sQnwinfo = sQnwinfo.substring("+QNWINFO:".length()).trim();
+            logger.trace("getQueryNetworkInformation() :: +QNWINFO={}", sQnwinfo);
+            asQnwinfo = sQnwinfo.split(",");
+            this.band = asQnwinfo[2].substring(1, asQnwinfo[2].length() - 1);
+            logger.trace("getBand() :: Band={}", this.band);
+            this.radio = asQnwinfo[0].substring(1, asQnwinfo[0].length() - 1);
+            logger.info("getRadio() :: Radio={}", this.radio);
+        }
+        return this.asQnwinfo;
+    }
+
+    public void getExtendedRegistrationStatus() throws KuraException {
+        synchronized (this.atLock) {
+            String atPort = getAtPort();
+
+            logger.debug("sendCommand getLac :: {}", QuectelGenericAtCommands.GET_REGISTRATION_STATUS.getCommand());
+            byte[] reply;
+            CommConnection commAtConnection = openSerialPort(atPort);
+            if (!isAtReachable(commAtConnection)) {
+                closeSerialPort(commAtConnection);
+                throw new KuraException(KuraErrorCode.NOT_CONNECTED, MODEM_NOT_AVAILABLE_FOR_AT_CMDS_MSG);
+            }
+            try {
+                commAtConnection.sendCommand(QuectelGenericAtCommands.GET_EXTENDED_REGISTRATION_STATUS.getCommand().getBytes(),
+                        1000, 100);
+                reply = commAtConnection.sendCommand(QuectelGenericAtCommands.GET_REGISTRATION_STATUS.getCommand().getBytes(),
+                        1000, 100);
+            } catch (IOException e) {
+                closeSerialPort(commAtConnection);
```

Comment:
we could close connection in a finally statement

Reply:
OK